### PR TITLE
Add support for only checking year of birth in LexisNexis (LG-3706)

### PR DESCRIPTION
**Pull request by zachmargolis**
Builds on: https://github.com/18F/identity-lexisnexis-api-client-gem/pull/20

- Adds a new `dob_year_only` top-level argument to the lambda payload body
- When enabled *and* we're checking for an AAMVA state, this:
    1. flips the order of checking so that AAMVA is first
    2. passes `dob_year_only` into LN, so that LN only sends DOB

Next steps:
1. Update the IDP to reference this and pass in the new param
2. Test that new param in staging (w/o background processing) to make sure this all actually works

### CODE REVIEW

File: source/proof_resolution/spec/proof_resolution_spec.rb [hunk 1]
```diff
@@ -204,6 +206,47 @@
       end
     end
 
+    context 'checking DOB year only' do
+      let(:dob_year_only) { true }
+
+      it 'only sends the birth year to LexisNexis (extra applicant attribute)' do
+        expect(aamva_proofer).to receive(:proof).and_return(Proofer::Result.new)
+        expect(lexisnexis_proofer).to receive(:proof).
+          with(hash_including(dob_year_only: true)).
```
Comment (stevegsa) [inline source/proof_resolution/spec/proof_resolution_spec.rb:215]: `dob_year_only: dob_year_only`

**Review by stevegsa — APPROVED** [review]
LGTM

**Review by mitchellhenke — APPROVED** [review]
👍🏼 

I don't think the IDP would/should have any issues, but will be keeping an eye out for whether switching the order of LN -> AAMVA to AAMVA -> LN in some cases causes issues.

**Comment by zachmargolis** [comment]
> I don't think the IDP would/should have any issues, but will be keeping an eye out for whether switching the order of LN -> AAMVA to AAMVA -> LN in some cases causes issues.

I agree, the ordering switch is the riskiest part of this, but I didn't see another way to accomplish the goal of the ticket

**Pull request merged**